presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -923,6 +923,33 @@ private void assertSTGeometryN(String wkt, int index, String expected)
         assertFunction("ST_ASText(ST_GeometryN(ST_GeometryFromText('" + wkt + "')," + index + "))", VARCHAR, expected);
     }
 
+    @Test
+    public void testLineString()
+    {
+        // General case, 2+ points
+        String query = "ST_LineString(array[ST_Point(1,2), ST_Point(3,4), ST_Point(5,6), ST_POINT(7,8)])";
+        String expected = "LINESTRING (1 2, 3 4, 5 6, 7 8)";
+        assertFunction(query, GEOMETRY, expected);
+
+        query = "ST_LineString(array[ST_Point(1,2), ST_Point(3,4), ST_Point(5, 6)])";
+        expected = "LINESTRING (1 2, 3 4, 5 6)";
+        assertFunction(query, GEOMETRY, expected);
+
+
+        query = "ST_LineString(array[ST_Point(1,2), ST_Point(3,4)])";
+        expected = "LINESTRING (1 2, 3 4)";
+        assertFunction(query, GEOMETRY, expected);
+
+        // Single point
+        query = "ST_LineString(array[ST_Point(9,10)])";
+        assertInvalidFunction(query, INVALID_FUNCTION_ARGUMENT);
```

Comment:
Could you check error message as well? 

`assertInvalidFunction(query, "LineString cannot have one point");`

Would you also test some other invalid inputs? E.g. `array[ST_GeometryFromText('POINT EMPTY')]`, `array[ST_GeometryFromText('POLYGON ...')]`, etc.